crates/net/network/src/transactions/mod.rs

```diff
@@ -1102,21 +1108,44 @@ where
                     metrics_increment_egress_peer_channel_full,
                 );
             }
-            // drain commands
+
+            // advance commands (propagate/fetch/serve txns)
             if let Poll::Ready(Some(cmd)) = this.command_rx.poll_next_unpin(cx) {
                 this.on_command(cmd);
                 some_ready = true;
             }
 
-            // drain incoming transaction events
+            // advance incoming transaction events (stream new txns/announcements from network
+            // manager and queue for import to pool/fetch txns)
+            //
+            // can potentially queue around 1,5 k txns for import to pool if txns are valid. each
+            // `Transactions` message could contain at least up to soft limit byte size for
+            // broadcast message / avg small legacy tx byte size: 128 KiB / 100 bytes < 1,5k
+            // transactions.
+            //
+            // if txns however are invalid, and just 1 byte, since this isn't validated until
+            // import to pool, this can potentially queue around > 130 k txns. more if the
+            // message size is bigger than 128 KiB.
+            //
+            // this will potentially remove hashes from hashes pending fetch, it the event is an
+            // announcement (if same hashes are announced that didn't fit into a previous request)
             if let Poll::Ready(Some(event)) = this.transaction_events.poll_next_unpin(cx) {
                 this.on_network_tx_event(event);
                 some_ready = true;
             }
 
             this.update_fetch_metrics();
 
-            // drain fetching transaction events
+            // advance fetching transaction events (flush transaction fetcher and queue for
```

Comment:
I have the same issue with this section, mentioning these specific values is not very helpful.

Instead, this should also start with the soft limit as the limiting factor and then an example calculation, this should be clear enough.

Reply:
it's very helpful for the developer to understand the proportions processed everywhere. I can see by how the future is implemented that this has not been taken into consideration before.